src/wallet/wallet.cpp

```diff
@@ -3717,6 +3737,9 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
             if (!walletInstance->SetHDMasterKey(masterPubKey))
                 throw std::runtime_error(std::string(__func__) + ": Storing master key failed");
         }
+        if (GetBoolArg("-disablehot", DEFAULT_DISABLE_HOT_WALLET)) {
+            walletInstance->AddWalletFlag(WALLET_FLAG_DISABLE_HOT_KEYS);
```

Comment:
This should also probably fake an encrypted wallet, so it can be read by older versions sanely.